src/charts/donut.js

```diff
@@ -282,18 +287,60 @@ define(function(require){
             svg.select('.donut-text').text('');
         }
 
+        /**
+         * Find the slice by id and growth it if needed
+         * @private
+         */
+        function initHighlightSlice() {
+            if (!highlightedSliceId && highlightedSliceId !== 0) return;
+
+            highlightedSlice = svg.selectAll('.chart-group .arc path')
+                .select(function(a) {
+                    if (a.data.id === highlightedSliceId) {
+                        return this;
+                    }
+                }).node();
+
+            if (!highlightedSlice) return;
```

Comment:
Nitpick: We would prefer
  if (highlightedSlice) {
     ...code
  }

thanks!